tests/test_bidi.py

```diff
@@ -850,6 +850,34 @@ async def test_scriptEvaluateWaitPromise_resultReturned(websocket):
         "value":"SOME_RESULT"
         }, resp["result"]["result"], ["objectId"])
 
+@pytest.mark.asyncio
+async def test_scriptEvaluateChangingObject_resultObjectDidNotChange(websocket):
+    contextID = await get_open_context_id(websocket)
+
+    # Send command.
+    await send_JSON_command(websocket, {
+        "id": 47,
+        "method": "script.evaluate",
+        "params": {
+            "expression": "(()=>{const a={i:0};const f=()=>{ setTimeout(()=>{ a.i++; f(); }, 0); };f();return a;})()",
```

Comment:
A comment explaining why the timing of this is guaranteed to be correct would be great. It has to be impossible for any subsequent CDP command to be handled before the `setTimeout` callback runs for it to be valid.

When there's an invoke method or something to serialize the value (given by objectId) again we could also check that the `value` becomes non-zero directly afterwards. (Nice-to-have.)